synapse/handlers/device.py

```diff
@@ -859,6 +859,7 @@ async def handle_room_un_partial_stated(self, room_id: str) -> None:
         known_hosts_at_join = await self.store.get_partial_state_servers_at_join(
             room_id
         )
+        assert known_hosts_at_join is not None
```

Comment:
Should this be an error? Usually assertions are used for programming errors, but this seem to be input validation?

Reply:
The assert can't (shouldn't?) ever be hit unless we've introduced a bug. This method is only run when we're about to finish syncing state. At that point the room is still partial stated, which implies we have a list of servers in it.